scala3doc-js/src/searchbar/engine/SearchbarEngine.scala

```diff
@@ -0,0 +1,9 @@
+package dotty.dokka
+
+import math.Ordering.Implicits.seqOrdering
+
+class SearchbarEngine(pages: List[PageEntry]) {
+  def query(query: List[Matchers]): List[PageEntry] = {
+    pages.map(p => p -> query.map(_(p))).filterNot(_(1).exists(_ < 0)).sortBy(_(1)).map(_(0))
```

Comment:
this is rather hard to understand, maybe we can split it into methods or multiple lines?

Also using code like `.sortBy{ case (_, measure) => measure}` helps a lot when working with traits

Reply:
You're right, little monster here. I've refactored this code to make it more readable.